### PR TITLE
read only method of running queries

### PR DESCRIPTION
**Pull request by parthshahva**
read-only user setup for query runner related to #220

### CODE REVIEW

File: scripts/redshift/setup_readonly_user.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+matrix_config = MatrixRedshiftConfig()
+redshift_handler = RedshiftHandler()
+
+
+def handler():
+    readonly_username = matrix_config.readonly_username
+    readonly_password = matrix_config.readonly_password
+    drop_user_query = f"DROP USER IF EXISTS {readonly_username};"
+    add_user_query = f"CREATE USER {readonly_username} WITH PASSWORD '{readonly_password}';"
+    grant_query = f"grant select on all tables in schema public to {readonly_username};"
```
Comment (mckinsel) [inline scripts/redshift/setup_readonly_user.py]: I think it needs select on `information_schema` to be able to return available fields dynamically.

Comment (mckinsel) [reply scripts/redshift/setup_readonly_user.py]: Well just `information_schema.tables` and `information_schema.columns`

Comment (parthshahva) [reply scripts/redshift/setup_readonly_user.py]: okay will grant select on all tables in information_schema

Comment (parthshahva) [reply scripts/redshift/setup_readonly_user.py]: will "grant select on all tables in schema information_schema" suffice?

Comment (mckinsel) [reply scripts/redshift/setup_readonly_user.py]: Yeah I think that should do it.

File: scripts/redshift/setup_readonly_user.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    redshift_handler.transaction([drop_user_query])
+    redshift_handler.transaction([add_user_query])
+    redshift_handler.transaction([grant_query])
```
Comment (calvinnhieu) [inline scripts/redshift/setup_readonly_user.py]: Single transaction?

File: matrix/common/aws/redshift_handler.py [hunk 3]
```diff
@@ -44,9 +44,16 @@ def __init__(self):
...
-    def transaction(self, queries: typing.List[str], return_results=False):
+    @property
+    def readonly_database_uri(self):
+        return self.redshift_config.readonly_database_uri
+
+    def transaction(self, queries: typing.List[str], return_results=False, read_only=False):
+        if read_only:
+            conn = pg.connect(self.readonly_database_uri)
+        else:
+            conn = pg.connect(self.database_uri)
```
Comment (calvinnhieu) [inline matrix/common/aws/redshift_handler.py]: Consider adding a unit test for these cases

**Pull request merged**